src/liballoc/string.rs

```diff
@@ -2106,26 +2106,19 @@ impl ops::DerefMut for String {
     }
 }
 
-/// An error when parsing a `String`.
+/// A type alias for [`Infallible`].
 ///
-/// This `enum` is slightly awkward: it will never actually exist. This error is
-/// part of the type signature of the implementation of [`FromStr`] on
-/// [`String`]. The return type of [`from_str`], requires that an error be
-/// defined, but, given that a [`String`] can always be made into a new
-/// [`String`] without error, this type will never actually be returned. As
-/// such, it is only here to satisfy said signature, and is useless otherwise.
+/// This alias exists for backwards compatibility, and will be eventually deprecated.
```

Comment:
Why will it be deprecated? Says who?

Reply:
If the never type works on all editions I don't see why people would ever use `Infalliable` long-term. Deprecation doesn't prevent its usage, just discourage it.